SU2_CFD/src/fluid/CNonPolytropicPengRobinson.cpp

```diff
@@ -0,0 +1,582 @@
+  fv = atanh;
+
+  sig = ComputeIntegralCp0_T_DT(T) - Gas_Constant * log(T / x2);
+  sres = - Gas_Constant + log((x2)/(x2 - b)) -
+         a * sqrt(alpha2(T)) * k * fv / (b * sqrt2 * sqrt(T * TstarCrit));
+
+  fx2 = sig + sres - s;
+
+  // zbrac algorithm NR
+
+  for (int j = 1; j <= NTRY; j++) {
+    if (fx1 * fx2 > 0.0) {
+      if (fabs(fx1) < fabs(fx2)) {
+        x1 += FACTOR * (x1 - x2);
+        T = T_P_rho(P, 1.0 / x1);
+
+        atanh = (log(1.0 + (b * sqrt2 / (x1 + b))) - log(1.0 - (b * sqrt2 / (x1 + b)))) / 2.0;
+        fv = atanh;
+        
+        sig = ComputeIntegralCp0_T_DT(T) - Gas_Constant * log(T / x1);
+        sres = - Gas_Constant + log((x1)/(x1- b)) -
+               a * sqrt(alpha2(T)) * k * fv / (b * sqrt2 * sqrt(T * TstarCrit));
+
+        fx1 = sig + sres - s;      
+        } else {
+        T = T_P_rho(P, 1.0 / x2);
+
+        atanh = (log(1.0 + (b * sqrt2 / (x2 + b))) - log(1.0 - (b * sqrt2 / (x2 + b)))) / 2.0;
+        fv = atanh;
+        
+        sig = ComputeIntegralCp0_T_DT(T) - Gas_Constant * log(T / x2);
+        sres = - Gas_Constant + log((x2)/(x2- b)) -
+               a * sqrt(alpha2(T)) * k * fv / (b * sqrt2 * sqrt(T * TstarCrit));
+
+        fx2 = sig + sres - s;  
+      }
+    }
+  }
+
+  // rtbis algorithm NR
+
+  f = fx1;
+  fmid = fx2;
+  if (f * fmid >= 0.0) {
+    cout << "Root must be bracketed for bisection in rtbis" << endl;
+    SetTDState_rhoT(Density, Temperature);
+  }
+  rtb = f < 0.0 ? (static_cast<void>(dx = x2 - x1), x1) : (static_cast<void>(dx = x1 - x2), x2);
+  do {
+    xmid = rtb + (dx *= 0.5);
+    T = T_P_rho(P, 1.0 / xmid);
+
+    atanh = (log(1.0 + (b * sqrt2 / (xmid + b))) - log(1.0 - (b * sqrt2 / (xmid + b)))) / 2.0;
+    fv = atanh;
+        
+    sig = ComputeIntegralCp0_T_DT(T) - Gas_Constant * log(T / xmid);
+    sres = - Gas_Constant + log((xmid)/(xmid- b)) -
+           a * sqrt(alpha2(T)) * k * fv / (b * sqrt2 * sqrt(T * TstarCrit));
+
+    fmid = sig + sres - s;
+
+    if (fmid <= 0.0) rtb = xmid;
+    count++;
+  } while (abs(fmid) > toll && count < ITMAX);
+
+  if (count == ITMAX) {
+    cout << "Too many bisections in rtbis" << endl;
+  }
+
+  rho = 1.0 / xmid;
+  T = T_P_rho(P, rho);
+  SetTDState_rhoT(rho, T);
+
+  cons_P = abs((Pressure - P) / P);
+  cons_s = abs((Entropy - s) / s);
+
+  if (cons_P > 1e-3 || cons_s > 1e-3) {
+    cout << "TD consistency not verified in hs call" << endl;
+  }
+}
+
+su2double CNonPolytropicPengRobinson::ComputeIntegralCp0_DT(su2double T){
+  su2double int_cp0 = 0;
+
+  for(size_t i_cp=0; i_cp < coeffs_.size(); i_cp++){
+    int_cp0 += coeffs_.at(i_cp) * pow(T, (i_cp + 1.0)) / (i_cp + 1.0);
+  }
+  return int_cp0;
+}
+
+su2double CNonPolytropicPengRobinson::ComputeIntegralCp0_T_DT(su2double T){
+  su2double int_cp0_T = 0;
+
+  for(size_t i_cp=0; i_cp < coeffs_.size(); i_cp++){
+
+    if (i_cp == 0) {
+      int_cp0_T += coeffs_.at(i_cp) * log(T);
+    } else {
+      int_cp0_T += coeffs_.at(i_cp) * pow(T, i_cp) / (i_cp);
```

Comment:
Do not use `at` on vectors or arrays, `at` checks bounds, here you are guaranteed to be within bounds.